src/charm.py

```diff
@@ -151,9 +154,18 @@ def reconcile(self, event: EventBase) -> None:
         if getattr(event, "departing_unit", None) == self.unit:
             return
 
-        # 2. Restart on config change
+        # 2. Restart if the service is down or on config change
+        service_stopped = not self.workload.alive() and not self.unit.status == MaintenanceStatus(
```

Comment:
In general we should avoid checking in the charm code the current status message. The check should rather be on the workload and **clear** the outdated status regardless of whether it was set or not.

Reply:
Actually, I cannot see where that `MSG_ROLLING_RESTART` is really set...

Reply:
So... that's the thing.

This is basically our only (one of the few) notations of the charm currently being in the restart process. This message is set by the `rolling_ops` lib.

However, as you are highligting it -- it's indeed not right this way.
(It's kind of a hack to rely on a status set outside of the charm's code, etc.)
Let me try to think of a better way.

Reply:
In fact, we don't need this check at all. Removed.